src/liballoc/collections/btree/node.rs

```diff
@@ -590,19 +592,13 @@ impl<'a, K: 'a, V: 'a, Type> NodeRef<marker::Mut<'a>, K, V, Type> {
     }
 
     fn into_key_slice_mut(mut self) -> &'a mut [K] {
```

Comment:
```suggestion
    /// The caller must ensure that the node is not the shared root.
    fn into_key_slice_mut(mut self) -> &'a mut [K] {
```
`into_val_slice` has such a comment, so seems only consistent to have it here, too.